src/pykeen/models/unimodal/complex.py

```diff
@@ -7,18 +7,44 @@
 import torch
 import torch.nn as nn
 
-from ..base import EntityRelationEmbeddingModel
+from ..base import InteractionFunction, SimpleVectorEntityRelationEmbeddingModel, normalize_for_einsum
 from ...losses import Loss, SoftplusLoss
 from ...regularizers import LpRegularizer, Regularizer
 from ...triples import TriplesFactory
-from ...utils import get_embedding_in_canonical_shape, split_complex
+from ...utils import split_complex
 
 __all__ = [
     'ComplEx',
+    'ComplexInteractionFunction',
 ]
 
 
-class ComplEx(EntityRelationEmbeddingModel):
+class ComplexInteractionFunction(InteractionFunction):
```

Comment:
should there be a different base interaction function for complex embeddings?

Reply:
to verify that we only apply interaction functions which regard the vector as complex number to complex numbers?

Reply:
I am actually curious to see whether we can switch to "real" complex vectors https://pytorch.org/docs/stable/generated/torch.complex.html, which would handle the type check / promotion for us.

Reply:
We can continue discussion on https://github.com/pykeen/pykeen/pull/134. As of now, this won't be possible because auto-differentiation on the `torch.nn.functional.embedding` is not possible for complex tensors.